### DISCUSSION TITLE
Setting stroke in BarSeries for xychart

### DISCUSSION THREAD

**Discussion by kjesle** [airbnb/visx#1605] [General]
Hello!

I'm trying to add a stroke around my bars in `BarSeries` in `@visx/xychart`. It seems like [BaseBarSeries](https://github.com/airbnb/visx/blob/bd96c7075cebe18a2f65f4168981240d33cf1d8c/packages/visx-xychart/src/components/series/private/BaseBarSeries.tsx#L87) does not return that as an option.

Are there other ways of setting the stroke for bars in a BarSeries or am I missing something?

**Comment by williaster** [comment]
hey @kjesle 👋  thanks for checking out visx. yikes, yes looks like some base svg props like `stroke` are not properly added in `BaseBarSeries` so we should get a fix out for this (I would note that the rounded bars are implemented with a `path` whereas the non-rounded/"typical" bars are implemented with a `rect` ... so we can't passthrough all e.g., all `rect` props.

fwiw, setting `stroke={string}` currently _works_, it just gives a type error. another alternative to styling is with the old-school css handle `.visx-bar`. [Sandbox link](https://codesandbox.io/s/bar-stroke-b39odu).

<img width="1340" alt="image" src="https://user-images.githubusercontent.com/4496521/206286523-79065378-a729-4b20-ab20-a23f2141c893.png">

**Comment by kjesle** [comment]
Thank you for a quick reply @williaster 

I forgot to mention that I use the `<BarSeries />` inside a `<BarGroup />`. Setting the fill and stroke manually there does nothing, but if I move the BarSeries it works.

I will continue with the inline css for now. Thanks!

**Comment by kjesle** [comment]
I had two `<BarSeries />` within a `<BarGroup />`, but there were no way of distinguish those two. Here is how I did it with inline css:

```
<style>{`
    .visx-bar-group > path:nth-child(n+${dataA.length + 1}) {
                        stroke: red;
                        fill: blue;
                    }      
    `}</style>
```
